Library/Homebrew/requirements/ruby_requirement.rb

```diff
@@ -33,4 +31,34 @@ def display_s
       name
     end
   end
+
+  private
+
+  def suitable_ruby
+    rubies.detect { |ruby| suitable?(ruby) }
+  end
+
+  def rubies
+    rubies = which_all("ruby")
+    if ruby_formula && ruby_formula.installed?
+      rubies.unshift Pathname.new(ruby_formula.bin/"ruby")
+    end
+    rubies.uniq
+  end
+
+  def suitable?(ruby)
+    version = Utils.popen_read(ruby, "-e", "print RUBY_VERSION").strip
+    version =~ /^\d+\.\d+/ && Version.create(version) >= min_version
+  end
+
+  def min_version
```

Comment:
It's used for every ruby in the loop. The previous implementation does `Version.create` for the minimum version on every iteration. Of course there's usually maybe 1..5 rubies so it's not a big deal.